src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -298,6 +300,454 @@ result<json::Document> parse_json(std::string_view v) {
     return {std::move(schema_json)};
 }
 
+/// is_superset section
+
+// a schema O is a superset of another schema N if every schema that is valid
+// for N is also valid for O. precondition: older and newer are both valid
+// schemas
+bool is_superset(json::Value const& older, json::Value const& newer);
+
```

Comment:
I think this forward declaration is not necessary

Reply:
yes it is right now. I put it here to implement the types that require recursion, like object and array. initially i thought to implement is_array_superset here but i decided to leave it for another pr